src/app/src/main/kotlin/org/vaccineimpact/api/app/ChunkedFileManager.kt

```diff
@@ -0,0 +1,51 @@
+package org.vaccineimpact.api.app
+
+import org.vaccineimpact.api.app.models.ChunkedFile
+import java.io.File
+import java.io.InputStream
+import java.io.RandomAccessFile
+
+open class ChunkedFileManager
+{
+    // Note: currentChunk is 1-indexed
+    fun writeChunk(inputStream: InputStream, contentLength: Int, metadata: ChunkedFile, currentChunk: Int)
+    {
+        File(UPLOAD_DIR).mkdir()
+        val uploadPath = "$UPLOAD_DIR/${metadata.uniqueIdentifier}.temp"
+        val raf = RandomAccessFile(uploadPath, "rw")
+
+        // Seek to position
+        raf.seek((currentChunk - 1) * metadata.chunkSize)
+
+        // Write to file
+        var readBytes: Long = 0
+        val bytes = ByteArray(1024 * 100)
+        while (readBytes < contentLength)
+        {
+            val r = inputStream.read(bytes)
+            if (r < 0)
+            {
+                break
+            }
+            raf.write(bytes, 0, r)
+            readBytes += r.toLong()
+        }
+        raf.close()
+    }
+
+    fun markFileAsComplete(chunkedFile: ChunkedFile)
```

Comment:
Tbh not sure if we really need this, we're not relying on filenames anywhere to determine whether a file is complete or not.

Reply:
exactly my thoughts, ok I'll leave it in :+1: